Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java

```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
+                    tts.shutdown();
+                    tts = null;
+
+                    Log.e(TAG, "TTS Language data is not available.");
+                    Toast.makeText(context, R.string.toast_error_tts_lang,
+                            Toast.LENGTH_LONG).show();
+                    break;
+            }
+
+            if (tts != null) {
+                tts.setOnUtteranceCompletedListener(mSpeechCompleteListener);
+
+                // Register the broadcast receiver
+                final IntentFilter intentFilter = new IntentFilter();
+                intentFilter.addAction(ACTION_SPEAK_MESSAGE);
+                intentFilter.addAction(SettingsFragment.ACTION_UPDATED_STATUS_PERIOD);
+
+                LocalBroadcastManager.getInstance(context).registerReceiver(
+                        mSpeechIntervalUpdateReceiver, intentFilter);
+
+                //Announce the connection event
+                watchdogCallback.setDrone(drone);
+                scheduleWatchdog();
+                speak(context.getString(R.string.speak_connected));
+            }
+        } else {
+            // Notify the user that the tts engine is not available.
+            Log.e(TAG, "TextToSpeech initialization failed.");
+            Toast.makeText(
+                    context,
+                    R.string.warn_tts_accessibility, Toast.LENGTH_LONG).show();
+        }
+    }
+
+    private void speak(String string) {
+        speak(string, false, null);
+    }
+
+    private void speak(String string, boolean append, String utteranceId) {
+        if (tts != null) {
+            if (shouldEnableTTS()) {
+                final int queueType = append ? TextToSpeech.QUEUE_ADD : TextToSpeech.QUEUE_FLUSH;
+
+                mTtsParams.clear();
+                if (utteranceId != null) {
+                    mTtsParams.put(TextToSpeech.Engine.KEY_PARAM_UTTERANCE_ID, utteranceId);
+                }
+
+                tts.speak(string, queueType, mTtsParams);
+            }
+        }
+    }
+
+    private boolean shouldEnableTTS() {
+        return mAppPrefs.isTtsEnabled();
+    }
+
+    private void speakArmedState(boolean armed) {
+        if (armed) {
+            speak(context.getString(R.string.speak_armed));
+        } else {
+            speak(context.getString(R.string.speak_disarmed));
+        }
+    }
+
+    private void batteryDischargeNotification(double battRemain) {
+        if (lastBatteryDischargeNotification > (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)
+                || lastBatteryDischargeNotification + 1 < (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)) {
+            lastBatteryDischargeNotification = (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT);
+            speak(context.getString(R.string.speak_battery_notification) + (int) battRemain + "%");
+        }
+    }
+
+    private void speakMode(VehicleMode mode) {
```

Comment:
@jandrop do these needs to be extracted as well? I'd imagine you'd want the same pronunciation for the flight modes regardless of the language.